go/vt/vtgate/vcursor_impl.go

```diff
@@ -1084,6 +1085,23 @@ func (vc *vcursorImpl) GetAggregateUDFs() []string {
 	return vc.vschema.GetAggregateUDFs()
 }
 
+// FindMirrorRule finds the mirror rule for the requested table name and
+// VSchema tablet type.
+func (vc *vcursorImpl) FindMirrorRule(name sqlparser.TableName) (*vindexes.MirrorRule, string, topodatapb.TabletType, key.Destination, error) {
+	destKeyspace, destTabletType, dest, err := vc.executor.ParseDestinationTarget(name.Qualifier.String())
+	if err != nil {
+		return nil, "", destTabletType, nil, err
+	}
+	if destKeyspace == "" {
+		destKeyspace = vc.keyspace
+	}
+	mirrorRule, err := vc.vschema.FindMirrorRule(destKeyspace, name.Name.String(), destTabletType)
+	if err != nil {
+		return nil, "", destTabletType, nil, err
+	}
+	return mirrorRule, destKeyspace, destTabletType, dest, err
+}
```

Comment:
should this not execute FindRoutedTable and then use that information to call FindMirrorRule?

Reply:
i wanted to avoid that because i didn't want routing rules to have an impact on mirror rules. i think it's lot simpler to think about, and less surprises this way.